Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -287,16 +318,18 @@ private void AttemptShoot(EntityUid user, EntityUid gunUid, GunComponent gun)
 
         if (attemptEv.Cancelled)
         {
-            if (attemptEv.Message != null)
+            if (attemptEv.Message != null && user != null)
             {
-                PopupSystem.PopupClient(attemptEv.Message, gunUid, user);
+                PopupSystem.PopupClient(attemptEv.Message, gunUid, user.Value);
             }
 
             gun.NextFire = TimeSpan.FromSeconds(Math.Max(lastFire.TotalSeconds + SafetyNextFire, gun.NextFire.TotalSeconds));
             return;
         }
 
-        var fromCoordinates = Transform(user).Coordinates;
+        // if the user exists, we use its coordinates. Otherwise - coordinates of the weapon.
+        var fromCoordinates = (user != null) ? Transform(user.Value).Coordinates : Transform(gunUid).Coordinates;
+
```

Comment:
As above the entitycoordinates change makes it difficult for a caller to tell what is happening.